x/gov/handler.go

```diff
@@ -30,39 +39,40 @@ func RegisterQuery(qr weave.QueryRouter) {
 func RegisterRoutes(r weave.Registry, auth x.Authenticator) {
 	propBucket := NewProposalBucket()
 	elecBucket := NewElectorateBucket()
-	r.Handle(pathVoteMsg, &VoteHandler{
+	r.Handle(pathVoteMsg, migration.SchemaMigratingHandler(packageName, &VoteHandler{
 		auth:       auth,
 		propBucket: propBucket,
 		elecBucket: elecBucket,
 		voteBucket: NewVoteBucket(),
-	})
-	r.Handle(pathTallyMsg, NewTallyHandler(auth, propBucket, elecBucket))
-	r.Handle(pathCreateTextProposalMsg, &TextProposalHandler{
+	}))
+	r.Handle(pathTallyMsg, migration.SchemaMigratingHandler(packageName,
+		NewTallyHandler(auth, propBucket, elecBucket)))
```

Comment:
All handlers are created in place and only this one has a constructor function :man_shrugging:

Reply:
yes, the others are simple